python/__init__.py

```diff
@@ -72,6 +72,27 @@ def __hash__(self):
 except ImportError:
     pass
 
+# Define a `with edit(layer)` statement
+
+class edit:
+    def __init__(self,layer):
+        self.layer = layer
+
+    def __enter__(self):
+        assert self.layer.startEditing()
+        return self.layer
+
+    def __exit__(self, ex_type, ex_value, traceback):
+        if ex_type is None:
+            assert self.layer.commitChanges()
```

Comment:
I would also check for the return error and we can return an exception here. `raise QgsFeatureExpection(errors)`.

```
try:
    with edit(layer):
           ....  
except QgsFeatureException:
    print "Doh!"
```